### PR TITLE
Support dynamic domains in mod_sic

### PR DESCRIPTION
**Pull request by Premwoik**
This PR adds support for dynamic domains in mod_sic and adapts tests.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71568) / small_tests / 90b7e48dd218c016403bd09e4c4e682bea4df652
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3258/71568/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3258/71568/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-10_07.26.03/lib.mongooseim.logs/run.2021-09-10_07.26.03/suite.log.html)

File: src/mod_sic.erl [hunk 1]
```diff
@@ -29,55 +29,77 @@
...
-    gen_iq_handler:remove_iq_handler(ejabberd_local, Host, ?NS_SIC),
-    gen_iq_handler:remove_iq_handler(ejabberd_sm, Host, ?NS_SIC).
+    [gen_iq_handler:add_iq_handler_for_domain(HostType, ?NS_SIC, Component, Fn, #{}, IQDisc) ||
+        {Component, Fn} <- iq_handlers()],
+    ok.
+
+-spec stop(mongooseim:host_type()) -> ok.
+stop(HostType) ->
+    [gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_LAST, Component) ||
+    {Component, _Fn} <- iq_handlers()],
```
Comment (chrzaszcz) [inline src/mod_sic.erl:63]: Minor: indentation

**Review by chrzaszcz — COMMENTED** [review]
The code looks good in general, I have one minor comment.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71680) / small_tests / 2108f0dcd1ea9f4142c60c422c4fae198a640c80
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3258/71680/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3258/71680/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-10_09.16.38/lib.mongooseim.logs/run.2021-09-10_09.16.39/suite.log.html)

File: src/mod_sic.erl [hunk 2]
```diff
@@ -29,55 +29,77 @@
...
+-ignore_xref([process_local_iq/5, process_sm_iq/5]).
 
 -include("jlib.hrl").
 -include("mongoose.hrl").
 -include("mongoose_config_spec.hrl").
 
 -define(NS_SIC, <<"urn:xmpp:sic:1">>).
 
-start(Host, Opts) ->
+-spec start(mongooseim:host_type(), list()) -> ok.
```
Comment (NelsonVides) [inline src/mod_sic.erl:52]:
```suggestion
-spec start(mongooseim:host_type(), gen_mod:module_opts()) -> ok.
```
🙂

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/71710) / small_tests / 6755787a6f0627f3564b4af4e2cfaef67a33153e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3258/71710/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3258/71710/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-10_10.12.17/lib.mongooseim.logs/run.2021-09-10_10.12.17/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Good!

**Pull request merged**